### PR TITLE
Replace any uses of .send().unwrap() by .send().unwrap_or_else().

### PR DESCRIPTION
**Pull request by asajeffrey**
This is part of the effort to harden the constellation (#10124), in this case against panics caused by channel send.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10242)

### CODE REVIEW

**Comment by metajack** [comment]
@bors-servo r+

This looks like it will give us much more information on these kinds of errors. That should be a definite improvement.

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/10242#-:-KDygN-TPl3z7aGi6Gfe:1808610558)*

**Comment by asajeffrey** [comment]
@metajack thanks!

**Comment by asajeffrey** [comment]
Hmm, that's two random git failures in a row... @bors-servo: retry

**Comment by jdm** [comment]
@bors-servo: retry
* #10251

File: components/compositing/constellation.rs [hunk 1]
```diff
@@ -665,16 +669,18 @@ impl<LTF: LayoutThreadFactory, STF: ScriptThreadFactory> Constellation<LTF, STF>
...
                     pipeline_id, event_type, button, point)) => {
-                if let Some(pipeline) = self.pipelines.get(&pipeline_id) {
-                    pipeline.script_chan.send(ConstellationControlMsg::SendEvent(pipeline_id,
-                        CompositorEvent::MouseButtonEvent(event_type, button, point))).unwrap();
-                }
+                let event = CompositorEvent::MouseButtonEvent(event_type, button, point);
+                let msg = ConstellationControlMsg::SendEvent(pipeline_id, event);
+                self.pipelines.get(&pipeline_id)
+                    .and_then(|pipeline| pipeline.script_chan.send(msg).ok())
+                    .unwrap_or_else(|| self.handle_send_error(pipeline_id));
```
Comment (nox) [inline components/compositing/constellation.rs]: FWIW I find this very wrong and I wouldn't r+. We have a tradition of not using and_then and unwrap_or_else for side-effectful code.

Comment (asajeffrey) [inline components/compositing/constellation.rs]: Discussion on IRC at http://logs.glob.uno/?c=mozilla%23servo#c395138.

**Comment by emilio** [comment]
BTW, I'd use `error!()`, or `warn!()` instead of `debug!()`, to know when a send fails in release.

**Comment by asajeffrey** [comment]
@emilio good point, I was just following the code that was there, but we should probably escalate the log level to at least `warn!`.

**Comment by asajeffrey** [comment]
Rebased and `git push --force` to see if this fixes the problem with appveyor git checkout (e.g. https://ci.appveyor.com/project/servo/servo/build/1.0.1416).

**Comment by asajeffrey** [comment]
@bors-servo: retry
* #10251

**Comment by asajeffrey** [comment]
@bors-servo retry
* "Test timed out"

**Pull request merged**